packages/jest-jasmine2/src/setup-jest-globals.js

```diff
@@ -48,59 +46,16 @@ const addSuppressedErrors = result => {
   }
 };
 
-function addAssertionErrors(result) {
-  const {
-    assertionCalls,
-    expectedAssertionsNumber,
-    isExpectingAssertions,
-  } = getState();
-  setState({
-    assertionCalls: 0,
-    expectedAssertionsNumber: null,
-  });
-  if (
-    typeof expectedAssertionsNumber === 'number' &&
-    assertionCalls !== expectedAssertionsNumber
-  ) {
-    const expected = EXPECTED_COLOR(
-      pluralize('assertion', expectedAssertionsNumber),
-    );
-    const message = new Error(
-      matcherHint('.assertions', '', String(expectedAssertionsNumber), {
-        isDirectExpectCall: true,
-      }) +
-        '\n\n' +
-        `Expected ${expected} to be called but only received ` +
-        RECEIVED_COLOR(pluralize('assertion call', assertionCalls || 0)) +
-        '.',
-    ).stack;
+const addAssertionErrors = result => {
+  const assertionErrors = extractExpectedAssertionsErrors();
+  if (assertionErrors.length) {
     result.status = 'failed';
-    result.failedExpectations.push({
-      actual: assertionCalls,
-      expected: expectedAssertionsNumber,
-      message,
-      passed: false,
-    });
-  }
-  if (isExpectingAssertions && assertionCalls === 0) {
-    const expected = EXPECTED_COLOR('at least one assertion');
-    const received = RECEIVED_COLOR('received none');
-    const message = new Error(
-      matcherHint('.hasAssertions', '', '', {
-        isDirectExpectCall: true,
-      }) +
-        '\n\n' +
-        `Expected ${expected} to be called but ${received}.`,
-    ).stack;
-    result.status = 'failed';
-    result.failedExpectations.push({
-      actual: 'none',
-      expected: 'at least one',
-      message,
+    result.failedExpectations = assertionErrors.map(error => ({
```

Comment:
you can't do this either :) you can potentially overwrite pre-existing test errors if you do this.

you should do:
```js
result.failedExpectations = result.failedExectations.concat(assertionErrors.map(error => {
  return {
    actual: error.assertionCalls,
    expected: error.expectedAssertionsNumber
    message: error.message,
    passed: false,
  };
}););
```

also you'll need to add `assertionCalls` and `expectedAssertionNumber` properties to the error object (inside `extractExpectedAssertionsErrors` module`)